### ISSUE TITLE
Oj.dump() generates invalid JSON when Float::NAN entry exists

### ISSUE THREAD

**Issue by tamayika** [ohler55/oj#779]
Ruby: 3.1.2
Oj: 3.13.14
OS: macOS Monterey

repro
```rb
require "oj"

puts Oj.dump({
  "float_test" => 0.25,
  "nan_test" => Float::NAN,
  "inf_test" => Float::INFINITY,
  "minus_inf_test" => -Float::INFINITY,
  "min_test" => Float::MIN,
  "max_test" => Float::MAX,
}) # => {"float_test":0.25,"nan_test":3.3e14159265358979323846
```

If `nan_test` entry is removed, output JSON is correct.

**Comment by ohler55** [comment]
What mode were you using and what is the expected result?

**Comment by tamayika** [comment]
No options are specified.
I just installed oj and ran repro code.
Repro code generates unterminated JSON and lacks fields after `nan_test`.

**Comment by ohler55** [comment]
I did some digging and found that in object mode the issue was present. It has been fixed in the develop branch. Please try and I'll release either tomorrow or when you indicate it is fixed which ever comes first.

**Comment by ohler55** [comment]
Released v3.13.15 with the fix.

**Comment by tamayika** [comment]
I've confirmed the issue was resolved in v3.13.15.
Thank you for the fix!
